### DISCUSSION TITLE
Manual "on the fly" address not linked to customer

### DISCUSSION THREAD

**Discussion by yoannisj** [craftcms/commerce#2405] [Ideas]
**Description**

When editing an order in the CP and adding a new address, this address gets saved in the database but it is not get linked to the order's customer (i.e. no new row saved in the `commerce_customers_addresses` db table).

FYI: strangely, the last created address does load up when I create a new order for the same customer.

**To Reproduce**
1. In the CP, go to /commerce/orders and click on "New Order"
2. Enter the email address of a customer in the "Customer" field
3. Click on the gear icon next to "Shipping Address" and select "New Address"
4. Fill-in address fields and confirm when done
5. Click on "Mark as complete" in the sidebar
6. Click on the "Update Order" button
7. Go to /commerce/customers and click on the customer with email entered in step 2.
8. The address is not listed under "Addresses"
9. The address id is not found in the `%commerce_customers_addresses` database

**Expected behavior**

The new customer address created when creating the new order in the CP should appear in the customer CP screen (i.e. admin/commerce/customers > customer email > addresses).

**Additional info**
- Craft version: 3.5.7
- Commerce version: 3.2.3
- PHP version: 7.4
- Database driver & version: Maria DB 1:10.2.30

**Comment by ghost** [comment]
Believe it or not, I just came to report exactly the same issue!
Craft Pro 3.5.8, Commerce 3.2.3, MySQL

**Comment by lukeholder** [comment]
Addresses that are added to a manually created order in the CP are not copied to the address book. We could look to add that feature. Marking this as a feature request.

**Comment by ghost** [comment]
That would be great, Luke, so thanks. The reason for this, by the way, is that the business I'm working with has traditionally taken a lot of orders through channels other than their new website, and that's going to continue for a while even as they focus on that website. In fact, they tell me that even then there will be a lot of customers that will probably never actually want to move to the web. They do want to use Commerce as their main order management solution and single source of truth, though.

So they have, and will continue to have, a lot of customers (including repeat customers) whose orders get entered manually. I know this is probably a bit unusual these days, but really—an order is an order, irrespective of where it comes from, so I think manual entries really do deserve to be integrated as completely as web-based ones. Things should be the same regardless of whether the client or the business creates the order.

On a related point, by the way, I have a suspicion that the Commerce dashboard widgets don't always include data from manual orders. Would that be something you know anything about?

**Comment by lukeholder** [comment]
Thanks for the feedback. Will keep that in mind.

> On a related point, by the way, I have a suspicion that the Commerce dashboard widgets don't always include data from manual orders. Would that be something you know anything about?

Shouldn't be the case, please raise a new ticket with steps to reproduce.

**Comment by Coultsy** [comment]
I find this very frustrating to explain to a client. In their mind addresses are stored against the customer. When they go to do a new order within admin and use the Craft look up a user/customer tool they can't understand why there is no address info even though the user has order before.
None of my clients use this 'New Order' tool because of this.
Would love it if there was an option to save address details to the customer as a part of the order process.

**Reply by lukeholder** [reply]
We could look to add `saveBillingAddressOnOrderComplete` which is available on the front-end to be available on the back-end.
